### PR TITLE
TT-5827: Add version suffix, allowing success key to prevent repeated…

### PR DESCRIPTION
**Pull request by tlevi**
Running migration over and over is silly (slow) and breaks easily.
Add a "success key" when migration is complete, keyed on a provided version suffix.

If REDLOCK_VERSION_SUFFIX is left unset it behaves the same.

Also:
* Version the migration-failed key too so that failures don't prevent old versions booting
* Increase delay for lock retry, no need for many nodes to hammer it so hard

### CODE REVIEW

File: lib/synchronised_migration/main.rb [hunk 1]
```diff
@@ -15,24 +15,43 @@ class << self
...
-    lock_and_execute
+    done_or_execute
   end
 
   private
 
+  def done_or_execute
+    return Result.ok if previous_success?
+    result = lock_and_execute
+    mark_successful if result.success?
```
Comment (blaknite) [inline lib/synchronised_migration/main.rb]: The rest of the mark/unmark calls are in `execute`. Would this call to `mark_successful` be better placed in `execute`?

File: .travis.yml [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
-script: bundle && bundle exec rspec
+rvm:
+  - 2.4
+  - 2.5
+  - 2.6
+script: bundle exec rspec
+sudo: false
+cache: bundler
+before_install:
+  - gem install bundler
```
Comment (Steffylicious) [inline .travis.yml]: https://stackoverflow.com/questions/44704968/visual-studio-code-insert-new-line-at-the-end-of-files

File: lib/synchronised_migration/main.rb [hunk 3]
```diff
@@ -15,24 +15,43 @@ class << self
...
     mark_failed
     migrate
-    return Result.new 'Migration failed.' if migration_failed?
+    return Result.fail 'Migration failed.' if migration_failed?
     remove_fail_marker
-    Result.new
+    return Result.ok
+  end
+
+  def previous_success?
```
Comment (blaknite) [inline lib/synchronised_migration/main.rb]: I was having difficulty mapping this method to the description of what it's doing and I think it is related to its name. This is returning success status for a given version yeah? If so, it's probably better if it is named after that use case.

**Review by Steffylicious — APPROVED** [review]

**Review by blaknite — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (tlevi) [reply .travis.yml]: Ugh you caught me leaning on rubocop

Comment (tlevi) [reply .travis.yml]: (Fixed)

**Review by blaknite — APPROVED** [review]

**Review by HellRok — APPROVED** [review]

**Pull request merged**